### ISSUE TITLE
addon-check fails to read http://mirrors.kodi.tv/addons/nexus/addons.xml.gz

### ISSUE THREAD

**Issue by ReenigneArcher** [xbmc/addon-check#272]
Since, approximately 2 weeks ago, this addon-check tool cannot read the above mentioned xml file.

I guess it's likely that the format of the file was modified, but it is rendering this checker useless.

The read is occurring here. https://github.com/xbmc/addon-check/blob/4f145d46cc9694489dce700fa7bd678370e26af2/kodi_addon_checker/addons/Repository.py#L33-L35

```txt
Traceback (most recent call last):
  File "C:\Users\ReenigneArcher\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ReenigneArcher\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Themerr-kodi\venv\Scripts\kodi-addon-checker.exe\__main__.py", line 7, in <module>
  File "C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Themerr-kodi\venv\lib\site-packages\kodi_addon_checker\__main__.py", line 86, in main
    all_repo_addons = check_addon.get_all_repo_addons()
  File "C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Themerr-kodi\venv\lib\site-packages\kodi_addon_checker\check_addon.py", line 133, in get_all_repo_addons
    repo_addons[branch] = Repository(branch, branch_url)
  File "C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Themerr-kodi\venv\lib\site-packages\kodi_addon_checker\addons\Repository.py", line 36, in __init__
    content = xml_file.read()
  File "C:\Users\ReenigneArcher\.pyenv\pyenv-win\versions\3.8.10\lib\gzip.py", line 292, in read
    return self._buffer.read(size)
  File "C:\Users\ReenigneArcher\.pyenv\pyenv-win\versions\3.8.10\lib\gzip.py", line 479, in read
    if not self._read_gzip_header():
  File "C:\Users\ReenigneArcher\.pyenv\pyenv-win\versions\3.8.10\lib\gzip.py", line 427, in _read_gzip_header
    raise BadGzipFile('Not a gzipped file (%r)' % magic)
gzip.BadGzipFile: Not a gzipped file (b'<h')
```

**Comment by ReenigneArcher** [comment]
I should have checked open PRs, but I guess this may be solved by #271

**Comment by razzeee** [comment]
Yes, that's the theory
